examples/bank/bank.go

```diff
@@ -0,0 +1,195 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Vivek Menezes (vivek.menezes@gmail.com)
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/security"
+	"github.com/cockroachdb/cockroach/security/securitytest"
+	"github.com/cockroachdb/cockroach/server"
+	"github.com/cockroachdb/cockroach/storage"
+	"github.com/cockroachdb/cockroach/testutils"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// Makes an id string from an id int.
+func makeAccountID(id int) []byte {
+	return []byte(fmt.Sprintf("%09d", id))
+}
+
+// Bank stores all the bank related state.
+type Bank struct {
+	kvClient        *client.KV
+	numAccounts     int
+	numTransactions int32
+}
+
+// moveMoney() moves an amount between two accounts if the amount
+// is available in the from account. Returns true on success.
+func (bank *Bank) moveMoney(from, to []byte, amount int64) bool {
+	// Early exit when from == to
+	if bytes.Compare(from, to) == 0 {
+		return true
+	}
+	txnOpts := &client.TransactionOptions{Name: fmt.Sprintf("Transferring %s-%s-%d", from, to, amount)}
+	err := bank.kvClient.RunTransaction(txnOpts, func(txn *client.Txn) error {
+		fromGet := client.Get(proto.Key(from))
+		fromResp := fromGet.Reply.(*proto.GetResponse)
+		toGet := client.Get(proto.Key(to))
+		toResp := toGet.Reply.(*proto.GetResponse)
+		if err := txn.Run(fromGet, toGet); err != nil {
+			return err
+		}
+		// Read from value.
+		var fromValue int64
+		if fromResp.Value != nil && fromResp.Value.Bytes != nil {
+			readValue, err := strconv.ParseInt(string(fromResp.Value.Bytes), 10, 64)
+			if err != nil {
+				return err
+			}
+			fromValue = readValue
+		}
+		// Ensure there is enough cash.
+		if fromValue < amount {
+			return nil
+		}
+		// Read to value.
+		var toValue int64
+		if toResp.Value != nil && toResp.Value.Bytes != nil {
+			readValue, err := strconv.ParseInt(string(toResp.Value.Bytes), 10, 64)
+			if err != nil {
+				return err
+			}
+			toValue = readValue
+		}
+		// Update both accounts.
+		txn.Prepare(client.Put(proto.Key(from), []byte(fmt.Sprintf("%d", fromValue-amount))))
+		txn.Prepare(client.Put(proto.Key(to), []byte(fmt.Sprintf("%d", toValue+amount))))
+		return nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	atomic.AddInt32(&bank.numTransactions, 1)
+	return true
```

Comment:
I think there is value to having the []byte stored in the value be returned via an API to the user instead of the user needing to deal with the proto message stuff.